athena-federation-sdk-dsv2/src/main/java/com/amazonaws/athena/connectors/dsv2/AthenaFederationAdapterDefinition.java

```diff
@@ -0,0 +1,129 @@
+/*-
+ * #%L
+ * athena-federation-sdk-dsv2
+ * %%
+ * Copyright (C) 2023 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connectors.dsv2;
+
+import com.amazonaws.athena.connector.lambda.domain.TableName;
+import com.amazonaws.athena.connector.lambda.handlers.MetadataHandler;
+import com.amazonaws.athena.connector.lambda.handlers.RecordHandler;
+import com.amazonaws.athena.connector.lambda.security.FederatedIdentity;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.amazonaws.services.securitytoken.model.GetCallerIdentityRequest;
+
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+// NOTE: Spark dsv2 adapters for Athena Federation must implement this class
+// Implementations of this class must also be java.io.Serializable
+public interface AthenaFederationAdapterDefinition
```

Comment:
An interface is more restrictive than an abstract class and therefore always preferred when you can express something as an interface instead of an abstract class.

For example, interfaces do not allow enclosing over any state, nor does it allow defining a constructor (because there's nothing to construct).

Interfaces are "more pure" in this sense.

Is there a reason you prefer an abstract class?